conjur/logic/credential_provider/keystore_credentials_provider.py

```diff
@@ -80,6 +81,16 @@ def remove_credentials(self, conjurrc):
         """
         Method for removing user credentials in the system's keyring
         """
-        logging.debug(f"Attempting to remove credentials from the '{KeystoreAdapter.get_keyring_name()}' credential store...")
-        KeystoreAdapter.delete_password(conjurrc.conjur_url)
-        logging.debug(f"Successfully removed credentials from the '{KeystoreAdapter.get_keyring_name()}' credential store...")
+        logging.debug(
+            f"Attempting to remove credentials from the '{KeystoreAdapter.get_keyring_name()}' credential store...")
+        for attr in KEYSTORE_ATTRIBUTES:
+            try:
+                KeystoreAdapter.delete_password(conjurrc.conjur_url, attr)
+            # Catches when credentials do not exist in the keyring. If the key does not exist,
+            # the user has already logged out. we still try to remove other leftovers
+            except KeyringDeletionError as err:
+                logging.debug(f"keyring unable to delete key {attr}. maybe not exist. {err}")
+                continue
```

Comment:
I don't think we want to continue we want to fail once and stop trying. I think we should `raise`

Reply:
`break`?

Reply:
i think this should be `return` as we had it before this change

Reply:
let's talk about this.

Reply:
as discussed in logout_controller we want to have a function that removes all left overs 
```
            if not self.credentials_provider.is_exists(loaded_conjurrc.conjur_url):
                # remove leftovers
                raise Exception("You are already logged out.")
```